ios/Codeforces Watcher/Application/components/CircleImageViewNew.swift

```diff
@@ -0,0 +1,40 @@
+import SwiftUI
+import SDWebImageSwiftUI
+
+struct CircleImageViewNew: View {
+    
+    private let noImageAvatarLink = "https://userpic.codeforces.org/no-avatar.jpg"
+    
+    let userAvatar: String
+    let borderColor: Color
+    
+    var body: some View {
+        if noImageAvatarLink == userAvatar {
+            Image("noImage")
+                .clipShape(Circle())
+                .overlay(
+                    Circle()
+                        .strokeBorder(borderColor, lineWidth: 1)
+                )
+        } else {
+            WebImage(url: URL(string: userAvatar))
```

Comment:
Won't just `WebImage` will have the same effect as `CircleImageViewNew` since it has a `placeholder(Image("noImage"))`?

Reply:
If the user doesn't have an avatar, we still get the default codeforces `no-avatar`, so the placeholder is only used when we try to get an image from a request.
So, the only solution I have found is to check if the user's avatar link is a link codeforces `no-avatar`.
Since we are currently using an image from the design to display the lack of an avatar.

Reply:
Got it, these are different images.

Reply:
If we use the default codeforces `no-avatar`, we don't need this condition.

Reply:
Nope, let's leave it as it is now.